modules/sku/src/AcquiaCommerce/SKUPluginBase.php

```diff
@@ -376,6 +385,10 @@ protected function getStock(SKU $sku, $reset = FALSE) {
 
     // Save the value in SKU if we came here as fallback of push mode.
     if ($stock_mode == 'push') {
+      if (!$sku instanceof SKU) {
+        $sku = SKU::loadFromSku($sku_string);
```

Comment:
Can you use entityManager for loading?

Reply:
loadFromSku uses static cache and takes care of translating to current language so using it everywhere, also we'll have sku string (not entity id) here